nucypher/characters/lawful.py

```diff
@@ -1748,34 +1735,28 @@ def interface_info_with_metadata(self):
     #
 
     def verify_kfrag_authorization(self,
-                                   alice: Alice,
-                                   kfrag: KeyFrag,
-                                   signed_writ: bytes,
-                                   work_order: 'WorkOrder'
+                                   hrac: bytes,
+                                   publisher: Alice,
+                                   authorized_kfrag: AuthorizedKeyFrag,
                                    ) -> VerifiedKeyFrag:
-        from nucypher.policy.orders import WorkOrder  # TODO: resolve ciruclar dependency
 
-        writ_hrac, writ_kfrag_checksum, writ_signature = WorkOrder.signed_writ_splitter(signed_writ)
-        reconstructed_writ = writ_hrac + writ_kfrag_checksum
+        # TODO: should it be a method of AuthorizedKeyFrag?
 
         try:
-            self.verify_from(alice, reconstructed_writ, signature=writ_signature)
+            self.verify_from(publisher, authorized_kfrag.writ, signature=authorized_kfrag.writ_signature)
         except InvalidSignature:
-            raise Policy.Unauthorized  # This isn't from Alice (relayer).
+            # TODO (#2740): differentiate cases for Policy.Unauthorized
+            raise Policy.Unauthorized  # This isn't from Alice (publisher).
 
-        if writ_hrac != work_order.hrac:  # Funky Workorder
+        if authorized_kfrag.hrac != hrac:  # Funky Workorder
             raise Policy.Unauthorized  # Bob, what the *hell* are you doing?
 
-        kfrag_checksum = keccak_digest(bytes(kfrag))[:WRIT_CHECKSUM_SIZE]
-        if kfrag_checksum != writ_kfrag_checksum:
-            raise Policy.Unauthorized  # Bob, Seriously?
-
         try:
-            verified_kfrag = kfrag.verify(verifying_pk=alice.stamp.as_umbral_pubkey())
+            verified_kfrag = authorized_kfrag.kfrag.verify(verifying_pk=publisher.stamp.as_umbral_pubkey())
```

Comment:
This reads a bit weird - similar to my comment in the `test_bob_handles_frags.py`. The "publisher" is used to verify kfrags...as opposed to alice? I know you mentioned there isn't a differentiation between the two at the moment, but naming the parameter "publisher" makes this weird.

Reply:
Yes, I fixed it - added a `provider` parameter to `verify_kfrag_authorization()`. It would be weird to call the parameter `alice` - do we have an accepted term for this? Is `provider` ok?

Reply:
Sorry if I'm missing something obvious here, but why is it weird to call this parameter "Alice"?  Isn't this always the KFrag generator?

Reply:
Well... we're confusing two naming schemes there. Publisher is named by its role in what's happening, and Alice is a character name. If it was Pablo and not Publisher that would have been at least consistent... Although to me, "Alice" is an instance and not a type, so to speak. When it was only one Alice at least it wasn't that bad, but now `alice` is `Alice`, and `publisher` is also `Alice`.